src/IR.cpp

```diff
@@ -821,6 +821,16 @@ bool Shuffle::is_interleave() const {
     return true;
 }
 
+int Shuffle::is_broadcast() const {
+    int lanes = vectors.front().type().lanes();
+    for (int i = 0; i < (int)indices.size(); i++) {
```

Comment:
Based on the use of is_broadcast, also need to check vectors.size() == 1 and indices.size() % vectors[0].type().lanes() == 0